s3/s3_operations.cpp

```diff
@@ -168,22 +169,22 @@ namespace irods_s3 {
         std::uint64_t thread_id = std::hash<std::thread::id>{}(std::this_thread::get_id());
 
         // ********* DEBUG - print L1desc for all
-        if (getRodsLogLevel() >= developer_messages_log_level) {
-            rodsLog(developer_messages_log_level, "%s:%d (%s) [[%lu]] ------------- L1desc ---------------\n",
+        if (irods::experimental::log::logger_config<s3_plugin_logging_category>::level >= log::level::debug) {
```

Comment:
Be careful checking the log level. It isn't thread-safe.

Reply:
The level only gets set on agent startup from the config in server_config.json.

The idea here was to skip the debug loop (over L1desc) if we are not logging debug logs.  Worst case as far as I can tell, if this test passed in error, we would iterate over L1desc but nothing would be logged because the actual log messages are set as debug.

Reply:
Cool with me.